pkg/prometheus/promcfg.go

```diff
@@ -421,6 +421,21 @@ func (cg *ConfigGenerator) MergeTLSConfigWithScrapeClass(tlsConfig *monitoringv1
 	return tlsConfig
 }
 
+func (cg *ConfigGenerator) GetScrapeClassExtraRelabelings(scrapeClass *monitoringv1.ScrapeClass) []*monitoringv1.RelabelConfig {
+	scrapeClassName := cg.defaultScrapeClassName
+
+	if scrapeClass != nil {
+		scrapeClassName = scrapeClass.Name
+	}
+
+	scrapeClass, found := cg.scrapeClasses[scrapeClassName]
+
+	if !found {
+		return nil
+	}
```

Comment:
Non-blocking!

This part is duplicated with the implementation of `MergeTLSConfigWithScrapeClass` and will probably repeat itself for every other field we add to ScrapeClasses. It would be nice if we could find a way of removing this duplication 🤔

Reply:
So I just noticed that we use this https://github.com/prometheus-operator/prometheus-operator/blob/e8d020e1d23a966432584839c3ac98aac9e18b42/pkg/prometheus/promcfg.go#L837 at the top of all configured resources and it returns nil if there is no scrape class so I simplified this PR's code and maybe the same should be done for the existing fields